### PR TITLE
Autotest: reduce dependency to mavproxy

### PR DESCRIPTION
**Pull request by khancyr**

### CODE REVIEW

**Comment by peterbarker** [comment]
I have a concern that because the `.tlog` produced my MAVProxy doesn't contain traffic from `--output` connection that our logs will start to become rather confusing.  The vehicle will appear to do things without being prompted, from the `.tlog` perspective.

**Comment by khancyr** [comment]
Need https://github.com/ArduPilot/MAVProxy/pull/528

**Comment by khancyr** [comment]
rebased

File: Tools/autotest/common.py [hunk 1]
```diff
@@ -1061,6 +1138,12 @@ def test_arm_feature(self):
         if not self.disarm_vehicle():
             self.progress("Failed to DISARM")
             raise NotAchievedException()
+        if not self.arm_vehicle_with_mavproxy():
```
Comment (peterbarker) [inline Tools/autotest/common.py]: Should we have a pattern of tests that we just do with mavproxy as `mavproxy_arm_vehicle`?

File: Tools/autotest/arducopter.py [hunk 2]
```diff
@@ -656,10 +656,8 @@ def fly_gps_glitch_loiter_test(self, timeout=30, max_distance=20):
         # initialise current glitch
         glitch_current = 0
         self.progress("Apply first glitch")
-        self.mavproxy.send('param set SIM_GPS_GLITCH_X %.7f\n' %
-                           glitch_lat[glitch_current])
-        self.mavproxy.send('param set SIM_GPS_GLITCH_Y %.7f\n' %
-                           glitch_lon[glitch_current])
+        self.set_parameter("SIM_GPS_GLITCH_X", glitch_lat[glitch_current])
```
Comment (peterbarker) [inline Tools/autotest/arducopter.py]: Need to add an epsilon

File: Tools/autotest/common.py [hunk 3]
```diff
@@ -414,13 +414,53 @@ def armed(self):
...
+    def arm_vehicle_with_mavproxy(self):
+        """Arm vehicle with mavlink arm message send from MAVProxy."""
         self.mavproxy.send('arm throttle\n')
         self.mav.motors_armed_wait()
         self.progress("ARMED")
         return True
 
-    def disarm_vehicle(self):
-        """Disarm vehicle with mavlink disarm message."""
+    def disarm_vehicle_with_mavproxy(self):
```
Comment (peterbarker) [inline Tools/autotest/common.py]: Maybe `mavproxy_disarm_vehicle`?

File: Tools/autotest/common.py [hunk 4]
```diff
@@ -616,16 +656,53 @@ def get_bearing(loc1, loc2):
...
+                                           1,
+                                           0,
+                                           0,
+                                           0,
+                                           0,
+                                           0,
+                                           0)
+            m = self.mav.recv_match(type='AUTOPILOT_VERSION',
+                                    blocking=True,
+                                    timeout=10)
```
Comment (peterbarker) [inline Tools/autotest/common.py]: Should be `self.get_sim_time() - tstart`

**Comment by OXINARF** [comment]
I think that doing the same test directly via MAVLink and via MAVProxy is too much... next we'll add other GCS to test them as well?

File: Tools/autotest/common.py [hunk 5]
```diff
@@ -616,16 +656,53 @@ def get_bearing(loc1, loc2):
...
-            raise NotAchievedException()
-        self.progress("AUTOPILOT_VERSION received")
+        tstart = self.get_sim_time()
+        while self.get_sim_time() - tstart < 10:
+            # Cannot use run_cmd otherwise the respond is lost during the wait for ACK
+            self.mav.mav.command_long_send(1,
+                                           1,
+                                           mavutil.mavlink.MAV_CMD_REQUEST_AUTOPILOT_CAPABILITIES,
+                                           0,  # confirmation
+                                           1,
```
[the remark this answers is not in the document]
Comment (khancyr) [reply Tools/autotest/common.py]: needed for request

File: Tools/autotest/common.py (hunk 3 above)
Comment (khancyr) [reply Tools/autotest/common.py]: done

File: Tools/autotest/common.py [hunk 6]
```diff
@@ -616,16 +656,53 @@ def get_bearing(loc1, loc2):
...
+                                           0)
+            m = self.mav.recv_match(type='AUTOPILOT_VERSION',
+                                    blocking=True,
+                                    timeout=10)
+            if m is not None:
+                self.progress("AUTOPILOT_VERSION received")
+                return
+        raise AutoTestTimeoutException()
 
     def do_set_mode_via_command_long(self):
```
[the remark this answers is not in the document]
Comment (khancyr) [reply Tools/autotest/common.py]: done

**Comment by OXINARF** [comment]
> However, continuing to test against MAVProxy means that we know
at least one GCS is working!  Also - sim_vehicle.py itself is based on
MAVProxy, so the tests do have some value there.

Proving that one GCS that mostly only devs use is working isn't very valuable - it's implementation was likely done by the same person that implemented the feature in ArduPilot.

sim_vehicle.py being based on MAVProxy is a wrong assumption though, you can launch SITL without launching MAVProxy, we added an option for that. It's debatable if we should launch MAVProxy by default, but given the lack of UDP support in SITL, it makes sense to me to have MAVProxy there (it totally confuses people on what is ArduPilot and what is MAVProxy though).

**Comment by khancyr** [comment]
@peterbarker solve the problem on quadplane ... It was a problem of timeout ... the run_cmd can be quite long waiting for the ack, so it just need to count time after getting the ACK ! ^^

File: Tools/autotest/common.py [hunk 7]
```diff
@@ -343,7 +343,8 @@ def clear_wp(self):
 
     def log_download(self, filename, timeout=360):
         """Download latest log."""
-        self.disarm_vehicle()
+        if self.armed():
```
Comment (peterbarker) [inline Tools/autotest/common.py]: log_download should never be called with the vehicle armed.

File: Tools/autotest/arduplane.py [hunk 8]
```diff
@@ -601,7 +602,6 @@ def autotest(self):
             self.progress("Home location: %s" % self.homeloc)
             self.wait_ready_to_arm()
             self.run_test("Arm features", self.test_arm_feature)
-            self.arm_vehicle()
```
Comment (peterbarker) [inline Tools/autotest/arduplane.py]: The vehicle should be armed within the run_flaps function.

File: Tools/autotest/arduplane.py [hunk 9]
```diff
@@ -508,7 +508,8 @@ def fly_flaps(self):
             self.mavproxy.send('switch 1\n')  # auto mode
             self.wait_mode('AUTO')
             self.wait_ready_to_arm()
-            self.arm_vehicle()
```
Comment (peterbarker) [inline Tools/autotest/arduplane.py]: This shouldn't be changed - arming the vehicle here makes it self-contained and you can rearrange the tests.

**Comment by khancyr** [comment]
Closed by 6e44ea13d4bb9013270394731d4ca450b0e629df

**Pull request closed without merging**